### PR TITLE
[#163] update linkButton

### PR DESCRIPTION
**Pull request by HsiuYa0**
- accept custom colors
- add faded press style
- add disable style and functionality

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -35,7 +35,7 @@
     "react-native-maps": "0.26.1",
     "react-native-paper": "^3.2.1",
     "react-native-picker-select": "^6.3.3",
-    "react-native-qrcode-svg": "^6.0.1",
+    "react-native-qrcode-svg": "^6.0.6",
```
Comment (jacksonrya) [inline package.json:38]: Did this change for a reason? If not, make sure to run npm install after reverting this file in order to ensure the package-lock.json file matches.

File: src/elements/LinkButton/LinkButton.tsx [hunk 2]
```diff
@@ -1,30 +1,62 @@
...
+			<TouchableOpacity
+				disabled={disabled}
+				activeOpacity={0.8}
+				onPress={buttonFunction}
+			>
+				<View
+					style={[
+						styles.textContainer,
+						{
+							backgroundColor: disabled
```
Comment (jacksonrya) [inline src/elements/LinkButton/LinkButton.tsx:40]: typo: `borderColor`

**Comment by jacksonrya** [comment]
For some reason, when I ran npm install with the corrected `package.json`, the package-lock file didn't change. So i reverted that file to it's version in master with `git reset 71c6cca package-lock.json` where `71c6cca` is the hash of the latest commit outside this PR.

One last thing, if you throw this code into the LoginScreen to debug:
```tsx
<LinkButton
	disabled={true}
	borderColor="pink"
	textColor="black"
	text="test button"
	onPress={() => { setNum(num + 1); }}
/>
```
notice that the colors revert to default if the button is disabled. The colors should remain the same, but the border color reverts to the default color

**Comment by HsiuYa0** [comment]
**usage example** 
example
```
<SafeAreaView>
    <LinkButton text="disabled button" disabled={true} />
    <LinkButton
	disabled={true}
	borderColor="pink"
	textColor="black"
	text="test button"
	onPress={() => { alert('button clicked'); }}
    />
</SafeAreaView>
```

**Comment by duskcloudxu** [comment]
Found an issue on IOS simulator, which would cause border to disappear(actually being set to "white") in login screen.
image
This problem is in LinkButton.styles.ts, I attached a snapshot below.
image
Your code use attributes "borderColor" to set border Color and in styles.ts, Abe used the "borderBottomColor".It's not guaranteed that the borderColor attribute would overwrite borderBottomColor, it differs from platforms (e.g. on the browser there is no such a problem). My suggestion is to delete two lines of code in the red box of snapshot above lol.

File: src/elements/LinkButton/LinkButton.tsx [hunk 3]
```diff
@@ -1,30 +1,62 @@
...
+			<TouchableOpacity
+				disabled={disabled}
+				activeOpacity={0.3}
+				onPress={buttonFunction}
+			>
+				<View
+					style={[
+						styles.textContainer,
+						{
+							borderColor: disabled && borderColor === colors.BANANA_YELLOW
```
Comment (jacksonrya) [inline src/elements/LinkButton/LinkButton.tsx:40]: Logic is still incorrect in order to achieve the correct disabled styling.
> disabled-link-button-review

1. The second button should have: text with disabled grey coloring.
2. The last button's border shouldn't change when disabled. No matter the state of the button, the border color should remain constant. Explore the Figma card in order to see how colors change between different button states. 

_Code to reproduce image_
```tsx
<LinkButton
	disabled={false}
	borderColor="pink"
	textColor="black"
	text="Enabled"
	onPress={() => { console.log('button clicked'); }}
/>

<LinkButton
	disabled={true}
	borderColor="pink"
	textColor="black"
	text="Disabled"
	onPress={() => { console.log('button clicked'); }}
/>

<LinkButton
	disabled={false}
	borderColor={BANANA_YELLOW}
	textColor={NAVY_BLUE}
	text="Enabled"
	onPress={() => { console.log('button clicked'); }}
/>

<LinkButton
	disabled={true}
	borderColor={BANANA_YELLOW}
	textColor={NAVY_BLUE}
	text="Disabled"
	onPress={() => { console.log('button clicked'); }}
/>
```

File: src/elements/LinkButton/LinkButton.styles.ts [hunk 4]
```diff
@@ -10,14 +10,12 @@ export default StyleSheet.create({
...
-		borderBottomColor: 'white',
 		borderBottomWidth: 3.5,
 	},
 	text: {
 		textAlign: 'center',
 		fontFamily: 'open-sans-bold',
 		fontSize: 20,
-		color: colors.NAVY_BLUE,
 	},
 	pressed: {
```
Comment (jacksonrya) [inline src/elements/LinkButton/LinkButton.styles.ts:20]: Unused style classes can be removed too :)

Comment (HsiuYa0) [reply src/elements/LinkButton/LinkButton.styles.ts]: removed :)

**Pull request merged**